app/component/project_card/project_card_component.html.erb

```diff
@@ -0,0 +1,32 @@
+<% if policy(@project).check_direct_view_access? %>
+  <div class="col-xs-12 col-sm-12 col-md-6 col-lg-4 users-circuits-column circuit-card-js">
+    <div class="card text-center users-card">
+      <div class="card-header users-card-header">
+        <div class="users-card-name">
+          <p><%= @project.name %></p>
+          <span class="tooltiptext"><%= @project.name %></span>
+        </div>
+        <div class="users-tag-container">
+          <% if !@project.assignment_id.nil? %>
+            <span class="badge search-tag"><%= t("assignment") %></span>
+          <% end %>
+          <span class="badge search-tag"><%= @project.project_access_type %></span>
+        </div>
+      </div>
+      <div class="card-body users-card-body">
+        <img class="users-card-image" src="<%= @project.image_preview.url %>" alt="<%= @project.name %>">
+      </div>
+      <div class="card-footer users-card-footer">
+        <% if policy(@project).user_access? %>
+          <a href="<%= simulator_edit_path(@project) %>" class="btn primary-button" target="_blank" rel="noopener noreferrer"><%= t("launch") %></a>
+        <% else %>
+          <%= link_to create_fork_project_path(@project), class: "btn primary-button", target: "_blank", method: :post, rel: "noopener" do %>
+            <span><%= t("fork") %></span>
+          <% end %>
+        <% end %>
+        <%= link_to t("view"), "#", class: "previewButton btn primary-button", data: { toggle: "modal", target: "#projectModal", currentproject: @project } %>
+        <a href="<%= user_project_path(@project.author_id, @project) %>" class="btn primary-button" target="_blank" rel="noopener noreferrer"><%= t("more") %></a>
+      </div>
+    </div>
+  </div>
+<% end %>
```

Comment:
@coderabbitai Improve this file by moving complexity from template to the component